### DISCUSSION TITLE
Remote respository for restic? (not a local mount point)

### DISCUSSION THREAD

**Discussion by willyp713** [DavidKrGH/BackupScripts#2] [General]
Since we already have rclone available through these scripts, I assume it wouldn't be too much trouble to use a remote repository instead of a local volume.

I don't have enough disk space to backup my entire system to a local disk first, and then later push to a remote storage with rclone.

Two thoughts I had about how this might work:
1. Use rclone to **mount a remote storage as a local volume** and then point the scripts to that
->Related example scripts in [drpoutine/rclone_mount](https://github.com/drpoutine/rclone_mount/tree/latest/Unraid_Scripts)
2. Initialize the **restic repository as a remote storage** - we never need to 'trick' the system into thinking it's a local mounted volume
->Example in the [Storj documentation](https://docs.storj.io/dcs/third-party-tools/restic)

Restic allows a direct connection to a remote repository with rclone all within the same `init` command like this:
`restic --repo rclone:storj:bucket/my-backup init`
...but there is so much else happening in these BackupScripts that there is probably more that would need to change.

**Comment by DavidKrGH** [comment]
Hello @willyp713,

In my opinion, both of the solutions you have described should be possible. Personally, however, I think the second approach is more effective, as you have to ensure that the connection is successfully established before each backup. This is then handled by the implementation directly in Restic.

I would also assume that the second approach requires fewer changes to the scripts. It should be sufficient to adapt/extend the three Restic executor files and the "BaseJob". As the script is modular, they can also be created in parallel.

If you are successful with your implementation, you are welcome to share your solution here. I may then extend the script for everyone with your approach.

Best regards.
